tools/getrealdeps.rb

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env ruby
+# getrealdeps for Chromebrew version 1.2
+# Author: Satadru Pramanik (satmandu) satadru at gmail dot com
+require 'fileutils'
+
+# Exit quickly if an invalid package name is given.
+if ARGV[0].nil? || ARGV[0].empty? || ARGV[0].include?('#')
+  puts 'Getrealdeps checks for the dependencies of a package.'
+  puts 'Dependencies are added if the package file is missing them.'
+  puts 'Usage: getrealdeps.rb <packagename>'
+  exit 1
+end
+
+pkg = ARGV[0].chomp('.rb')
+
+CREW_PREFIX = ENV['CREW_PREFIX'] || `crew const CREW_PREFIX`.split('=')[1].chomp
+CREW_LIB_PREFIX = ENV['CREW_LIB_PREFIX'] || `crew const CREW_LIB_PREFIX`.split('=')[1].chomp
+
+# Package needs to be installed for package filelist to be populated.
+unless File.exist? "#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist"
+  puts "Installing #{pkg} because it is not installed."
+  system("yes | crew install #{pkg}")
+end
+
+unless File.exist?("#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist")
+  puts "Package #{pkg} either does not exist or does not contain any libraries."
+  exit 1
+end
+
+# Speed up grep.
+ENV['LC_ALL'] = 'C'
+
+# Install grep if a functional local copy does not exist.
+if system('grep --version > /dev/null 2>&1')
+  GREP = 'grep'
+else
+  system('crew install grep')
+  GREP = "#{CREW_PREFIX}/bin/grep"
+end
+
+# Gawk is needed for adding dependencies.
+unless system('gawk -W version > /dev/null 2>&1')
+  puts "\nGawk is needed for adding dependencies to package files."
+  system('crew install gawk')
+end
+
+# upx is needed to expand compressed binaries to check for dependencies.
+unless system('upx --version > /dev/null 2>&1')
```

Comment:
If we've gotten to this stage in the build process and we have compressed binaries, we must have used upx to compress them right before, so we're guaranteed to have it

Reply:
This can be run independently of the build though...

Reply:
We check for upx just in case, but if run standalone, buildessential may not be installed.